ibmcloud/image/multipart_upload.sh

```diff
@@ -0,0 +1,61 @@
+
+function usage() {
+    echo "Usage: $0 --file <file name> --bucket <bucket name> [--size <size of parts 20M-100M>]"
+}
+
+# cleanup, removes the split files and the structure if they have been created by this script
+function cleanup() {
+    rm -f ${original_file}_structure.json
+    rm -f $original_file.*
+}
+
+while (( "$#" )); do
+    case "$1" in
+        --file) original_file=$2 ;;
+        --bucket) bucket=$2 ;;
+        --size) split_size=$2 ;;
+        --help) usage; exit 0 ;;
+        *)      usage 1>&2; exit 1;;
+    esac
+    shift 2
+done
+
+# Default to 100M part size
+split_size=${split_size:-100M}
+
+# Requires file and bucket to be specified to continue
+if [[ -z "${original_file-}" || -z "${bucket-}"  ]]; then
+    usage 1>&2
+    exit 1
+fi
+
+# Divide the original file up into smaller sections
+split $original_file -b $split_size $original_file.
+# Create a multipart-upload, need the upload-id for part uploads 
+upload_id=$(ibmcloud cos multipart-upload-create --bucket $bucket --key $original_file --output JSON | jq -r '.UploadId')
+if [[ -z $upload_id ]]; then 
+    echo "Unable to start upload, check permissions" 1>&2
+    cleanup
```

Comment:
You don't need to call `cleanup` explicitly at every possible exit.

trap 'cleanup "image creation failed"' 0

Instead, you can use `trap` to define a signal handler like this.

```
trap cleanup EXIT
```
`EXIT` here means that the handler is executed on exit from the shell.